paper-api/src/main/java/org/bukkit/block/BlockType.java

```diff
@@ -169,6 +170,16 @@ interface Typed<B extends BlockData> extends BlockType {
         @Override
         B createBlockData();
 
+        /**
+         * Creates a set of {@link BlockData} instances for this block type, with all
+         * possible combinations of properties values.
+         *
+         * @return new block data set
+         */
+        @NotNull
+        @Override
+        Set<B> createBlockDataStates();
```

Comment:
The return type shouldn't be `Set`. The contents are mutable objects which can break set invariants. The type shouldn't be `List` either because the order and random access shouldn't be used, so I think Collection is the best option.